File: backend/src/main/java/com/springbootjwtpostgres/backend/order/OrderSearchCriteria.java
package com.springbootjwtpostgres.backend.order;

import lombok.Data;

import java.util.Date;
import java.util.List;

@Data
public class OrderSearchCriteria {
    private List<OrderType> orderTypes;
    private List<OrderStatus> orderStatuses;
    private double orderTotalPriceFrom;
    private double orderTotalPriceTo;
    private Date createdAtFrom;
    private Date createdAtTo;
}

File: frontend/src/app/models/order/OrderSearchCriteria.ts
import { EOrderType } from './EOrderType';
import { EOrderStatus } from './EOrderStatus';

export class OrderSearchCriteria {
  orderTypes?: EOrderType[];
  orderStatuses?: EOrderStatus[];
  orderTotalPriceFrom?: number;
  orderTotalPriceTo?: number;
  createdAtFrom?: Date;
  createdAtTo?: Date;
}
